### PR TITLE
legacy train fix

### PR DESCRIPTION
**Pull request by cdoern**
we should only shutil the files from input-dir to data-dir if data-dir DNE. The way this is happening now makes it so that you copy the same files on top of one another resulting in a `SameFileError`

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/model/train.py [hunk 1]
```diff
@@ -360,6 +360,7 @@ def get_files(directory: str, pattern: str) -> list[str]:
         # The first file is latest
         logger.debug("train_file=%s", train_files[0])
         logger.debug("test_file=%s", test_files[0])
+        print(train_files)
```
Comment (RobotSail) [inline src/instructlab/model/train.py:363]: Did you mean to include this?

Comment (cdoern) [reply src/instructlab/model/train.py]: removed

File: src/instructlab/model/train.py [hunk 2]
```diff
@@ -311,13 +311,13 @@ def train(
     if four_bit_quant and device != "cuda":
         ctx.fail("'--4-bit-quant' option requires '--device=cuda'")
 
-    effective_data_dir = Path(DEFAULTS.DATASETS_DIR)
+    effective_data_dir = Path(data_path or DEFAULTS.TAXONOMY_DIR)
```
Comment (RobotSail) [inline src/instructlab/model/train.py:314]: Are we still going to output the generated files into the taxonomy directory?

**Comment by cdoern** [comment]
@RobotSail is taking over this PR

File: src/instructlab/model/train.py (hunk 2 above)
Comment (nathan-weinberg) [reply src/instructlab/model/train.py]: Has this been addressed?

Comment (RobotSail) [reply src/instructlab/model/train.py]: Yeah, I moved it back to be DATASETS_DIR but kept the conditional in the case when `data_path` is provided.

**Review by nathan-weinberg — APPROVED** [review]

**Review by alinaryan — APPROVED** [review]

**Pull request merged**